worker/uniter/relation/statetracker.go

```diff
@@ -498,15 +498,21 @@ func (r *relationStateTracker) Report() map[string]interface{} {
 	result := make(map[string]interface{})
 
 	for id, st := range r.stateMgr.(*stateManager).relationState {
-		relationer := r.relationers[id]
-		result[strconv.Itoa(id)] = map[string]interface{}{
+		report := map[string]interface{}{
```

Comment:
It seems quite strange for relationState to include the id of a relation for which there isn't an entry in relationers.

Reply:
Indeed, I believe this shows an indicative issue of the internal state. The panic doesn't actually help to try to locate the cause of the inconsistent state, but hopefully, this can be a start.